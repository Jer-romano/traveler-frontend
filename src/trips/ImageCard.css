.ImageCard img {
    width: 500px;
}