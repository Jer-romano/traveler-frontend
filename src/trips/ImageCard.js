import React from 'react';
import "./ImageCard.css";
import Tag from './Tag';

const ImageCard = ({ image, caption, tag1, tag2, tag3, tag4, tag5 }) => {

    console.debug("ImageCard:", image);
    return (
        <div className='card mt-3'>
            <div className='ImageCard card-body'>
                <div>
                    <img src={image} alt='Trip image' />
                </div>      
                <span className='card-body mt-3'>
                    <Tag tag={tag1}/>
                    <Tag tag={tag2}/>
                    <Tag tag={tag3}/>
                    <Tag tag={tag4}/>
                    <Tag tag={tag5}/>
                </span>      
                <h4 className='mt-3'> {caption} </h4>
            </div>
        </div>
     
    );

};

export default ImageCard;