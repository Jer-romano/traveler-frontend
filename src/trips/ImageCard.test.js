import React from "react";
import { render } from "@testing-library/react";
import ImageCard from "./ImageCard";

it("renders without crashing", function() {
    const path = "../images/brazil_photo2.jpeg"
    render(<ImageCard
                image={path}
                caption={"hello"}
                tag1={"good"}
                tag2={"bad"}
                tag3={"ugly"}/>);
});

it("matches snapshot", function() {
    const path = "../images/brazil_photo2.jpeg"
    const { asFragment } = render(
        <ImageCard
            image={path}
            caption={"hello"}
            tag1={"good"}
            tag2={"bad"}
            tag3={"ugly"}/>
    );
    expect(asFragment()).toMatchSnapshot();
})
