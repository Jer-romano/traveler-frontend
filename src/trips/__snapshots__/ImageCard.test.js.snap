// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card mt-3"
  >
    <div
      class="ImageCard card-body"
    >
      <div>
        <img
          alt="Trip image"
          src="../images/brazil_photo2.jpeg"
        />
      </div>
      <span
        class="card-body mt-3"
      >
        <span
          class="tag"
        >
          good
        </span>
        <span
          class="tag"
        >
          bad
        </span>
        <span
          class="tag"
        >
          ugly
        </span>
      </span>
      <h4
        class="mt-3"
      >
         hello 
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
